crypto/sm2/sm2_crypt.c

```diff
@@ -95,7 +95,7 @@ int sm2_ciphertext_size(const EC_KEY *key, const EVP_MD *digest, size_t msg_len,
     if (field_size == 0 || md_size < 0)
         return 0;
 
-    *ct_size = 12 + 2 * field_size + (size_t)md_size + msg_len;
+    *ct_size = 20 + 2 * field_size + (size_t)md_size + msg_len;
```

Comment:
So why 20? I'm not a fan of "magic" numbers. It will be useful to have a comment explaining where these numbers come from.

Reply:
The function above also has such magic number 10, and this was also 10 initially.

Reply:
In long form length octets, there is how 20 comes from:
[Sequence Tag, 1 byte] + [Sequence Length Size, 1 byte] + [Sequence Length, max to 4 byte*]
[Integer Tag, 1 byte] + [Integer Length, 1 byte] + [00 prefix, 1 byte] + [X Data]
[Integer Tag, 1 byte] + [Integer Length, 1 byte] + [00 prefix, 1 byte] + [Y Data]
[String Tag, 1 byte] + [String Length, 1 byte] + [C3 Data]
[String Tag, 1 byte] + [String Length Size, 1 byte] + [String Length, max to 4 byte*] + [C2 Data]

> * means, we assume that data in field is smaller than 3GB

So, 6+3+3+2+6 = 20

Reply:
I don't see why 20 either.  A 3GB size means four bytes of length so I could understand 16.  Where is the other 4 coming from?

Reply:
"Sequence Length" is the length of all followed data, from [Integer Tag, 1 byte] to [C2 Data]. Sequence Length is four bytes when C2 Data is four bytes.

Reply:
What about using `ASN1_object_size` to calculate the true length ?

Reply:
Thus:

```
    size_t sz;

    sz = 2 * ASN1_object_size(0, field_size + 1, V_ASN1_OCTET_STRING)
         + ASN1_object_size(0, md_size, V_ASN1_INTEGER)
         + ASN1_object_size(0, msg_len, V_ASN1_INTEGER);
    *ct_size = ASN1_object_size(0, sz, V_ASN1_SEQUENCE);
```

Reply:
Sounds great!  but it should be:

Reply:
Thanks, that's great! but md and msg are string, x and y field are integer. I think it should be:
```
size_t sz;

sz = 2 * ASN1_object_size(0, field_size + 1, V_ASN1_INTEGER)
         + ASN1_object_size(0, md_size, V_ASN1_OCTET_STRING)
         + ASN1_object_size(0, msg_len, V_ASN1_OCTET_STRING);
*ct_size = ASN1_object_size(0, sz, V_ASN1_SEQUENCE);
```

PR is updated

Reply:
Apologies for getting the types backwards.

Something similar might be possible for the case above that starts `10 +`